File: python_t4_full/Clone_2878.py
def add(self, val) :
	if (self.root == None) :
		self.root = Node(val)
	else :
		self._add(val, self.root)


def add(self, val):
      if (not self.root):
        self.root = Node(val)
      else:
        def _add(node, val):
          if (val < node.data):
            if not node.left:
              node.left = Node(val)
            else:
              _add(node.left, val)
          elif (val > node.data):
            if not node.right:
              node.right = Node(val)
            else:
              _add(node.right, val)
        _add(self.root, val)

File: cs_t4_full/Clone_2878.cs
private static void updateBandwidthInterval (double [] xval, int i, int [] bandwidthInterval) {
    int left = bandwidthInterval [0];
    int right = bandwidthInterval [1];
    int nextRight = nextNonzero (weights, right);
    if (nextRight < xval.Length && xval [nextRight] - xval [i] < xval [i] - xval [left]) {
        int nextLeft = nextNonzero (weights, bandwidthInterval [0]);
        bandwidthInterval [0] = nextLeft;
        bandwidthInterval [1] = nextRight;
    }
}




private static void updateBandwidthInterval (double [] xval, int i, int [] bandwidthInterval) {
    int nextRight = nextNonzero (weights, bandwidthInterval [1]);
    if (nextRight < xval.Length) {
        double temp = xval [nextRight] - xval [i];
        if(temp < xval [i] -xval [bandwidthInterval [0]]) {
            bandwidthInterval [0] = nextNonzero (weights, bandwidthInterval [0]);
            bandwidthInterval [1] = nextRight;
        }
    }   
}


